tiberius/src/query.rs

```diff
@@ -298,6 +299,11 @@ impl QueryRow {
         let col_data = &self.0.columns[idx];
         R::from_column_data(col_data).map(Some)
     }
+    pub fn iter(
```

Comment:
This exposes MetaDataColumn, which is an implementation specific representation.
Same as above applies.
Exposing this should be done using a NewType wrapper instead that exposes methods for data that we'll want and won't change (e.g. name).
Also I think we should use a slice for that instead of an iterator (similar to the postgres/mysql implementation).